crates/miden-lib/src/account/interface/mod.rs

```diff
@@ -0,0 +1,360 @@
+            .mast_forest()
+            .procedure_digests()
+            .all(|proc_digest| procedures.contains_key(&proc_digest))
+        {
+            basic_fungible_faucet_library().mast_forest().procedure_digests().for_each(
+                |component_procedure| {
+                    procedures.remove(&component_procedure);
+                },
+            );
+
+            component_interface_vec.push(AccountComponentInterface::BasicFungibleFaucet);
+        }
+
+        // RPO Falcon 512
+        // ------------------------------------------------------------------------------------------------
+
+        let rpo_falcon_proc = rpo_falcon_512_library()
+            .mast_forest()
+            .procedure_digests()
+            .next()
+            .expect("rpo falcon 512 component should export exactly one procedure");
+
+        if let Some(proc_info) = procedures.remove(&rpo_falcon_proc) {
+            component_interface_vec
+                .push(AccountComponentInterface::RpoFalcon512(proc_info.storage_offset()));
+        }
+
+        // Custom interfaces
+        // ------------------------------------------------------------------------------------------------
+
+        let mut custom_interface_procs_map = BTreeMap::<u8, Vec<AccountProcedureInfo>>::new();
+        procedures.into_iter().for_each(|(_, proc_info)| {
+            match custom_interface_procs_map.get_mut(&proc_info.storage_offset()) {
+                Some(proc_vec) => proc_vec.push(*proc_info),
+                None => {
+                    custom_interface_procs_map.insert(proc_info.storage_offset(), vec![*proc_info]);
+                },
+            }
+        });
+
+        if !custom_interface_procs_map.is_empty() {
+            for proc_vec in custom_interface_procs_map.into_values() {
+                component_interface_vec.push(AccountComponentInterface::Custom(proc_vec));
+            }
+        }
+
+        component_interface_vec
+    }
+
+    // probably it would be convenient to also have `from_components` constructor
+}
+
+// ACCOUNT COMPONENT INTERFACE
+// ================================================================================================
+
+/// Describes whether a note is compatible with a specific account.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum NoteAccountCompatibility {
+    /// A note is incompatible with an account.
+    ///
+    /// The account interface does not have procedures for being able to execute at least one of
+    /// the program execution branches.
+    No,
+    /// The account has all necessary procedures of one execution branch of the note script. This
+    /// means the note may be able to be consumed by the account if that branch is executed.
+    Maybe,
+}
+
+// HELPER FUNCTIONS
+// ================================================================================================
+
+/// Returns a vector of digests of all account component interfaces.
+pub fn component_proc_digests(
```

Comment:
Do we ever use this function outside of `AccountInterface`? If not, we could just change this into a method on `AccountInterface` - e.g., `AccountInterface::get_procedure_digests()`.

Also, this can probably return `BTreeSet<Digest>` right?

Reply:
It can also be private or `pub(super)` I think.

Reply:
We use it inside the `well_known_note.rs`, but I think it is a good idea to make it a method of the `AccountInterface`. And because of this usage we can't make it private, but `public(crate)` at most.